src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +38,28 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                int dotnetMonitorPayloadProtocolVersion;
+                long payloadLengthBuffer;
+                byte[] payloadBuffer;
+
+                using (var reader = new BinaryReader(StdInStream, Encoding.UTF8, true))
+                {
+                    dotnetMonitorPayloadProtocolVersion = reader.ReadInt32();
+                    payloadLengthBuffer = reader.ReadInt64();
+
+                    payloadBuffer = new byte[payloadLengthBuffer];
+
+                    reader.Read(payloadBuffer, 0, (int)payloadLengthBuffer); // this should be safe, given we know the payload is small
+                }
+
+                if (dotnetMonitorPayloadProtocolVersion != ExpectedPayloadProtocolVersion)
```

Comment:
This check needs to be moved to immediately after reading the protocol version. There's no point in attempting to read anything else until it verifies the expected protocol.